crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -3107,6 +3107,67 @@ impl<'db> TypeInferenceBuilder<'db> {
                     dataclass_params.is_some_and(|params| params.contains(DataclassParams::FROZEN))
                 };
 
+                let setattr_dunder_call_result = object_ty.try_call_dunder_with_policy(
+                    db,
+                    "__setattr__",
+                    &mut CallArgumentTypes::positional([
+                        Type::StringLiteral(StringLiteralType::new(db, Box::from(attribute))),
+                        value_ty,
+                    ]),
+                    MemberLookupPolicy::MRO_NO_OBJECT_FALLBACK,
+                );
+
+                let is_assignable_via_setattr = match setattr_dunder_call_result {
+                    Ok(res) => match res.return_type(db) {
+                        Type::Never => {
```

Comment:
one issue here is `Never` got [introduced in 3.11](https://docs.python.org/3/library/typing.html#typing.Never)

maybe match `NoReturn` as well? (3.6)

Reply:
You don't need to worry about that here. `Type::Never` is a version-independent representation of the bottom type in Rust, and it includes both `typing.Never` and `typing.NoReturn`.